### ISSUE TITLE
pypdf hangs on one specific PDF - cannot catch error with try/except

### ISSUE THREAD

**Issue by amplicity** [py-pdf/pypdf#1825]
pypdf freezes on one specific PDF. I get no error, and no succeed. I am not able to see what the error is.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
macOS-12.3.1-arm64-arm-64bit

$ python -c "import pypdf;print(pypdf.__version__)"
3.8.1
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from pypdf import PdfReader

reader = PdfReader("example.pdf")
number_of_pages = len(reader.pages)
page = reader.pages[0]
text = page.extract_text()
```
The PDF is shared below. This is a publicly-accessible PDF.

[MiFO_LFO_FEIS_NOA_Published (3).pdf](https://github.com/py-pdf/pypdf/files/11367871/MiFO_LFO_FEIS_NOA_Published.3.pdf)

Any thoughts as to why this particular pdf might hang?

**Comment by MartinThoma** [comment]
The fix was just merged and will be part of `pypdf >= 3.9.0`. It will be released today. A regression test was added.

**Comment by MartinThoma** [comment]
@amplicity Thank you for reporting the issue. If you want, I can add you as a contributor: https://pypdf.readthedocs.io/en/latest/meta/CONTRIBUTORS.html
